tests/benchmark/compute/instruction/test_system.py

```diff
@@ -67,6 +68,10 @@ def test_xcall(
     max_contract_size = fork.max_code_size()
 
     gas_costs = fork.gas_costs()
+    tx_gas_limit_cap = fork.transaction_gas_limit_cap()
+    assert tx_gas_limit_cap is not None, (
+        "This benchmark requires a tx gas limit cap"
+    )
```

Comment:
Could we mark this test for Osaka forward only? (Or maybe that already happens since AFAIK in all the benchmark releases only Prague was filled, since I'm not interested we want backwards performance runs. For zkVM also makes sense that previous forks aren't interesting for this attack).

I think we could make it work backwards, it would just blow up a the logic below reg tx preparations for setup and attack. This test is already quite complex so not sure it is worth it.

Reply:
```suggestion
    tx_gas_limit_cap = fork.transaction_gas_limit_cap() or gas_benchmark_value
```

I believe this should be sufficient to support both pre-Osaka and post-Osaka scenarios. If `fork.transaction_gas_limit_cap()` returns `None` (e.g., in Prague), it will fall back to `gas_benchmark_value`, and the transaction preparation logic will result in a single transaction:

```python
full_txs = attack_gas_limit // tx_gas_limit_cap   # becomes 1
remainder = attack_gas_limit % tx_gas_limit_cap   # becomes 0
```

Reply:
The setup phase also requires some touching since should use `env.gas_limit` instead of `tx_gas_limit_cap`; that's mainly what I didn't like that much if not going to be used, but I can do it if feels worthwhile. I'll try and report later.

Reply:
@LouisTsai-Csie, generalized it in the last commit. Tested both again (Prague and Osaka) and looks fine, so now it works on any fork.